### PR TITLE
User agent normalization

### PR DESCRIPTION
**Pull request by LucGenetier**
Add userAgent to PowerPlatformConnectorClient to generate a User Agent when calling APIM like below
`MyProduct/v1.2 PowerFx/1.3.0-build.20240110-1001`

Following RFC9110

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Connectors/PowerPlatformConnectorClient.cs [hunk 1]
```diff
@@ -42,23 +44,90 @@ public class PowerPlatformConnectorClient : HttpClient
...
+        /// <param name="userAgent">Product UserAgent to add to Power-Fx one (Power-Fx/version).</param>
+        /// <param name="httpInvoker">Optional HttpMessageInvoker. If not provided a default HttpClient is used.</param>
+        public PowerPlatformConnectorClient(string endpoint, string environmentId, string connectionId, Func<Task<string>> getAuthToken, string userAgent, HttpMessageInvoker httpInvoker = null)
         {
             _client = httpInvoker ?? new HttpClient();
 
             GetAuthToken = getAuthToken ?? throw new ArgumentNullException(nameof(getAuthToken));
             ConnectionId = connectionId ?? throw new ArgumentNullException(nameof(connectionId));
             EnvironmentId = environmentId ?? throw new ArgumentNullException(nameof(environmentId));
+            UserAgent = string.IsNullOrWhiteSpace(userAgent) ? $"PowerFx/{Version}" : $"{userAgent} PowerFx/{Version}";
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/PowerPlatformConnectorClient.cs:130]: Just double checking if this is the scheme (with a space). 
Separating by space seems weird.

Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Connectors/PowerPlatformConnectorClient.cs:130]: We follow RFC9110 here and spaces are OK

User-Agent = product *( RWS ( product / comment ) )
RWS = 1*( SP / HTAB )

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**